lib/dartdoc.dart

```diff
@@ -254,16 +254,36 @@ class DartDoc {
     }
 
     // Use the includeExternals.
-    for (Source source in context.librarySources) {
+    bool matchesLibrary(String includeExternal, Source source) {
       LibraryElement library = context.computeLibraryElement(source);
       String libraryName = Library.getLibraryName(library);
-      var fullPath = source.fullName;
-      if (includeExternals.any((string) => fullPath.endsWith(string))) {
-        if (libraries.map(Library.getLibraryName).contains(libraryName)) {
-          continue;
-        }
-        libraries.add(library);
+      return source.fullName.endsWith(includeExternal);
+    }
+
+    for (String includeExternal in includeExternals) {
+      var quotedArgString = "'--include-external $includeExternal'";
+      List<Source> matchedSources = context.librarySources
+          .where((library) => matchesLibrary(includeExternal, library));
```

Comment:
What would it mean to re-iterate the iterable?  Is this a general pattern we should have the analyzer warn about?